### PR TITLE
added 2 odin aws detections

### PR DESCRIPTION
**Pull request by hbenac10**
### Background

Added 2 odin detection rules

### Changes
New rules:
aws_cloudtrail_forge_web_credentials.py
aws_cloudtrail_unsuccessful_mfa_attempt.py
aws_ami_modified_for_public_access.py
aws_ami_modified_for_public_access.yml

### CODE REVIEW

File: rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from panther_base_helpers import deep_get, aws_rule_context
+
+def rule(event):
+    mfa_used = deep_get(event,"additionalEventData","MFAUsed")
```
Comment (edyesed) [inline rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py:4]: this needs to provide a default value as well, looks like the empty string would be appropriate here

File: rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    console_login = deep_get(event,"responseElements","ConsoleLogin")
```
Comment (edyesed) [inline rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py:5]: this deep_get should provide a default value, the empty string is probably good

File: rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from panther_base_helpers import deep_get, aws_rule_context
+
+def rule(event):
```
Comment (edyesed) [inline rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py:3]: I believe this rule should provide `eventSource` and `eventName` filters before going into the evaluation below.

File: rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    mfa_used = deep_get(event,"additionalEventData","MFAUsed")
+    console_login = deep_get(event,"responseElements","ConsoleLogin")
+
+    if mfa_used == "Yes" and console_login == "Failure":
+        return True
+    return False
+
+def title(event):
+    return f'Failed MFA login from {deep_get(event,"userIdentity", "userName")}'
```
Comment (edyesed) [inline rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py:12]: please use `userIdentity.arn` to refer to IAM originators

File: rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            accountId: "111122223333"
+            principalId: AIDACKCEVSQ6C2EXAMPLE
+            type: IAMUser
+            userName: anaya
+      Name: Unsuccessful Login w/ MFA
+DedupPeriodMinutes: 15
+LogTypes:
+    - AWS.CloudTrail
+RuleID: Unsuccessful.MFA.attempt
+Threshold: 2
```
Comment (edyesed) [inline rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.yml:70]: awesome! 2 MFA fails in 15 minutes == alert

File: rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            accessKeyId: ""
+            accountId: "111122223333"
+            principalId: AIDACKCEVSQ6C2EXAMPLE
+            type: IAMUser
+            userName: anaya
+      Name: Unsuccessful Login w/ MFA
+DedupPeriodMinutes: 15
+LogTypes:
+    - AWS.CloudTrail
+RuleID: Unsuccessful.MFA.attempt
```
Comment (edyesed) [inline rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.yml:69]: Can you please preface the RuleID with `AWS.` . There's a bit of a mismatch between how ruleIDs are generated in the web-ui and how we use them in this repo

File: rules/aws_cloudtrail_rules/aws_cloudtrail_forge_web_credentials.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
+from panther_base_helpers import deep_get, aws_rule_context
```
Comment (edyesed) [inline rules/aws_cloudtrail_rules/aws_cloudtrail_forge_web_credentials.py:1]: In the current form, this detection is going to alert on every successful AUTH_EVENTS. I believe this detection needs to be revisited to alert on these events when they're interesting in a security context. I'm not sure offhand how to identify when they're interesting vs when they're not

**Review by edyesed — CHANGES_REQUESTED** [review]

File: rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py (hunk 3 above)
Comment (hbenac10) [reply rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py:3]: This has been addressed.

File: rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py (hunk 1 above)
Comment (hbenac10) [reply rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.py]: This has been addressed.

File: rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.yml (hunk 5 above)
Comment (hbenac10) [reply rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.yml:70]: Thanks

File: rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.yml (hunk 6 above)
Comment (hbenac10) [reply rules/aws_cloudtrail_rules/aws_cloudtrail_unsuccessful_mfa_attempt.yml]: This has been addressed.

File: rules/aws_cloudtrail_rules/aws_cloudtrail_forge_web_credentials.py (hunk 7 above)
Comment (hbenac10) [reply rules/aws_cloudtrail_rules/aws_cloudtrail_forge_web_credentials.py:1]: See asana task: https://app.asana.com/0/0/1203074962567709/1203509321364067/f

Comment (edyesed) [reply rules/aws_cloudtrail_rules/aws_cloudtrail_forge_web_credentials.py:1]: @hbenac10 please remove this detection from the PR for the time being.

Comment (hbenac10) [reply rules/aws_cloudtrail_rules/aws_cloudtrail_forge_web_credentials.py]: Removed files.

**Review by edyesed — DISMISSED** [review]
Adds a detection for when a user tries to login with an MFA and receives a login failure. The threshold for alerting is 2 failed logins in a 15 minute period

**Review by edyesed — APPROVED** [review]

**Pull request merged**